File: backend/create_admin.py
import sys
import os
import getpass
sys.path.insert(0, os.path.dirname(__file__))

from passlib.context import CryptContext
from app.core.database import SessionLocal
from app.models.user import User
from datetime import datetime, timezone

pwd_context = CryptContext(schemes=['bcrypt'], deprecated='auto')
db = SessionLocal()

def get_admin_password():
    password = os.getenv('ADMIN_PASSWORD')
    if password:
        return password
    try:
        password = getpass.getpass('请输入管理员密码: ')
        confirm = getpass.getpass('请确认密码: ')
        if password != confirm:
            print('错误: 两次输入的密码不一致')
            sys.exit(1)
        if len(password) < 8:
            print('错误: 密码长度至少8位')
            sys.exit(1)
        return password
    except (EOFError, KeyboardInterrupt):
        print('\n错误: 未提供密码')
        sys.exit(1)

try:
    admin_password = get_admin_password()
    existing = db.query(User).filter(User.username == 'admin').first()
    if existing:
        print('Admin用户已存在，更新密码...')
        existing.hashed_password = pwd_context.hash(admin_password)
        existing.role = 'admin'
        existing.is_active = True
        db.commit()
        print('密码已更新')
    else:
        admin = User(
            username='admin',
            email='admin@youding.com',
            hashed_password=pwd_context.hash(admin_password),
            display_name='管理员',
            role='admin',
            is_active=True,
            created_at=datetime.now(timezone.utc),
            updated_at=datetime.now(timezone.utc),
        )
        db.add(admin)
        db.commit()
        print('Admin用户创建成功!')
        print('用户名: admin')
except Exception as e:
    db.rollback()
    print(f'错误: {e}')
finally:
    db.close()

File: seo-backend/src/scripts/createAdmin.js
require('dotenv').config();
const bcrypt = require('bcryptjs');
const { sequelize, models } = require('../config/database');
const AdminUser = models.AdminUser;

const createAdmin = async () => {
  try {
    console.log('开始创建管理员账号...');
    
    await sequelize.authenticate();
    console.log('数据库连接成功');

    const existingUser = await AdminUser.findOne({ where: { username: 'admin' } });
    
    if (existingUser) {
      console.log('管理员账号已存在，跳过创建');
      process.exit(0);
    }

    const password = process.env.ADMIN_PASSWORD || 'admin@123';
    const hashedPassword = await bcrypt.hash(password, 10);

    const adminUser = await AdminUser.create({
      username: 'admin',
      nickname: '超级管理员',
      passwordHash: hashedPassword,
      email: 'admin@example.com',
      status: 1,
      createdAt: new Date(),
      updatedAt: new Date()
    });

    console.log('管理员账号创建成功！');
    console.log(`用户名: admin`);
    console.log(`密码: ${password}`);
    console.log(`用户ID: ${adminUser.id}`);
    
    process.exit(0);
  } catch (error) {
    console.error('创建管理员账号失败:', error);
    process.exit(1);
  }
};

createAdmin();